composables/useCharactersFilters.ts

```diff
@@ -0,0 +1,41 @@
+import { QueryParams } from "~/constants/query-params";
+
+export function useCharactersFilters () {
+  const route = useRoute();
+  const router = useRouter();
+
+  const filters = computed({
```

Comment:
it looks not bad but it's not typed yet. 
you should define the params you expect 

`computed<{ status?: string, ...etc  }>`